include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
 	api_register_func('api/oauth/request_token', 'api_oauth_request_token', false);
 	api_register_func('api/oauth/access_token', 'api_oauth_access_token', false);
 
-	function api_fr_photos_list($type) {
-		if (api_user() === false) {
+
+	/**
+	 * @brief delete a complete photoalbum with all containing photos from database through api
+	 *
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_fr_photoalbum_delete($type) {
+		if (api_user()===false)
 			throw new ForbiddenException();
+
+		// input params
+		$album = (x($_REQUEST,'album') ? $_REQUEST['album'] : "");
+
+		// we do not allow calls without album string
+		if ($album == "")
+			throw new BadRequestException("no albumname specified");
+
+		// check if album is existing
+		$r = q("SELECT DISTINCT `resource-id` FROM `photo` WHERE `uid` = %d AND `album` = '%s'",
+				intval(api_user()),
+				dbesc($album));
+		if (!dbm::is_result($r))
+			throw new BadRequestException("album not available");
+
+		// function for setting the items to "deleted = 1" which ensures that comments, likes etc. are not shown anymore
+		// to the user and the contacts of the users (drop_items() performs the federation of the deletion to other networks
+		foreach ($r as $rr) {
+			$photo_item = q("SELECT `id` FROM `item` WHERE `uid` = %d AND `resource-id` = '%s' AND `type` = 'photo'",
+						intval(local_user()),
+						dbesc($rr['resource-id'])
+						);
+
+			if (!dbm::is_result($photo_item))
+				throw new InternalServerErrorException("problem with deleting items occured");
+
+			drop_item($photo_item[0]['id'],false);
 		}
 
-		$r = q("SELECT `resource-id`, MAX(`scale`) AS `scale`, `album`, `filename`, `type`
-				FROM `photo`
-				WHERE `uid` = %d AND `album` != 'Contact Photos' GROUP BY `resource-id`, `album`, `filename`, `type`",
+		// now let's delete all photos from the album
+		$result = q("DELETE FROM `photo` WHERE `uid` = %d AND `album` = '%s'",
+				intval(api_user()),
+				dbesc($album));
+
+		// return success of deletion or error message
+		if ($result) {
+			$answer = array('result' => 'deleted', 'message' => 'album `' . $album . '` with all containing photos has been deleted.');
+			return api_format_data("photoalbum_delete", $type, array('$result' => $answer));
+		}
+		else {
+			throw new BadRequestException("unknown error");
```

Comment:
this should never be reached, because a missing dataset in table is catched in row 3313. This was just added to avoid the warning that not every code path returns a value. However, I will change it to "unknown error - deleting from database failed".